### PR TITLE
chore: apply pep8-naming rules for naming convention

### PR DESCRIPTION
**Pull request by bowenliang123**
# Checklist:

> [!IMPORTANT]  
> Please review the checklist below before submitting your pull request.

- [x] Please open an issue before creating a PR or link to an existing issue
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I ran `dev/reformat`(backend) and `cd web && npx lint-staged`(frontend) to appease the lint gods

# Description

- apply the Ruff's pep8-naming rules `N`: https://docs.astral.sh/ruff/rules/#pep8-naming-n
- pep8-naming rules benefits in alignments to PEP8 with:
  - enforce correct usage of`@classmethod` `@staticmethod` decorators along with their first argument (N804 rule)
     - for the validators of Pydantic, `@classmethod` is placed after `@model_validator` and `@field_validator` as suggested by Pydantic docs ` ... The first argument should be cls (and we also recommend you use @classmethod below @model_validator for proper type checking), ... `  (https://docs.pydantic.dev/latest/concepts/validators/#model-validators)
  - enforce CapWords convention in class names
  - enforce Exception name {name} should be named with an Error suffix (N818)

- add new globally exclude rules:
  - N806 is too strict to enforcing all the variable names in lowercase. We have common usage in getting parameters in tool parameters, which often contains CapWord style.
  - N815 is excluded as we have `en_US` style available in classes.
## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update, included: [Dify Document](https://github.com/langgenius/dify-docs)
- [x] Improvement, including but not limited to code refactoring, performance optimization, and UI/UX improvement
- [ ] Dependency upgrade

# Testing Instructions

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: api/core/model_runtime/model_providers/volcengine_maas/legacy/volc_sdk/__init__.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
 from .common import ChatRole
-from .maas import MaasException, MaasService
+from .maas import MaasExceptionError, MaasService
```
Comment (laipz8200) [inline api/core/model_runtime/model_providers/volcengine_maas/legacy/volc_sdk/__init__.py]: Do you want to rename it to `MaasError`?

File: api/core/ops/entities/config_entity.py [hunk 2]
```diff
@@ -26,6 +26,7 @@ class LangfuseConfig(BaseTracingConfig):
     host: str = "https://api.langfuse.com"
 
     @field_validator("host")
+    @classmethod
```
Comment (laipz8200) [inline api/core/ops/entities/config_entity.py:29]: Why did you remove this `@classmethod`?

File: api/core/ops/entities/config_entity.py [hunk 3]
```diff
@@ -45,6 +46,7 @@ class LangSmithConfig(BaseTracingConfig):
     endpoint: str = "https://api.smith.langchain.com"
 
     @field_validator("endpoint")
+    @classmethod
```
Comment (laipz8200) [inline api/core/ops/entities/config_entity.py:49]: Why did you remove this `@classmethod`?

**Review by laipz8200 — CHANGES_REQUESTED** [review]

File: api/core/model_runtime/model_providers/volcengine_maas/legacy/volc_sdk/__init__.py (hunk 1 above)
Comment (bowenliang123) [reply api/core/model_runtime/model_providers/volcengine_maas/legacy/volc_sdk/__init__.py]: OK, fixed. Changed to `MaasError`.

File: api/core/ops/entities/config_entity.py (hunk 2 above)
Comment (bowenliang123) [reply api/core/ops/entities/config_entity.py:29]: Hi, @classmethod decorator is added in this PR, rather than removing it.

File: api/core/ops/entities/config_entity.py (hunk 3 above)
Comment (bowenliang123) [reply api/core/ops/entities/config_entity.py:49]: as above.

File: api/core/ops/entities/config_entity.py (hunk 2 above)
Comment (laipz8200) [reply api/core/ops/entities/config_entity.py:29]: Sorry, my mistake.

File: api/core/indexing_runner.py [hunk 4]
```diff
@@ -78,8 +78,8 @@ def run(self, dataset_documents: list[DatasetDocument]):
                     dataset_document=dataset_document,
                     documents=documents,
                 )
-            except DocumentIsPausedException:
-                raise DocumentIsPausedException("Document paused, document id: {}".format(dataset_document.id))
+            except DocumentIsPausedExceptionError:
```
Comment (laipz8200) [inline api/core/indexing_runner.py]: The same error

File: api/core/llm_generator/output_parser/errors.py [hunk 5]
```diff
@@ -1,2 +1,2 @@
-class OutputParserException(Exception):
+class OutputParserExceptionError(Exception):
```
Comment (laipz8200) [inline api/core/llm_generator/output_parser/errors.py]: The same error.

**Review by laipz8200 — CHANGES_REQUESTED** [review]

File: api/pyproject.toml [hunk 6]
```diff
@@ -47,9 +47,10 @@ ignore = [
     "B006", # mutable-argument-default
     "B007", # unused-loop-control-variable
     "B026", # star-arg-unpacking-after-keyword-arg
-#    "B901", # return-in-generator
     "B904", # raise-without-from-inside-except
     "B905", # zip-without-explicit-strict
+    "N806", # non-lowercase-variable-in-function
+    "N815", # mixed-case-variable-in-class-scope
```
Comment (bowenliang123) [inline api/pyproject.toml:53]: Some clarification for these globally exclued rules:
- N806 is too strict to enforcing all the variable names in lowercase. We have common usage in getting parameters in tool parameters, which often contains CapWord style.
- N815 is excluded as we have `en_US` style available in classes.

File: api/core/llm_generator/output_parser/errors.py (hunk 5 above)
Comment (bowenliang123) [reply api/core/llm_generator/output_parser/errors.py]: Thanks, fixed.

File: api/core/indexing_runner.py (hunk 4 above)
Comment (bowenliang123) [reply api/core/indexing_runner.py]: Thanks, fixed all similar errors.

Comment (bowenliang123) [reply api/core/indexing_runner.py]: We now have no occurance of `ExceptionError` appendix in class names.

**Review by laipz8200 — APPROVED** [review]
LGTM

**Comment by bowenliang123** [comment]
Thanks !
